modules/social_features/social_group/social_group.api.php

```diff
@@ -51,18 +51,36 @@ function hook_social_group_default_visibility_alter(&$visibility, $group_type_id
   }
 }
 
+/**
+ * Provide a method to alter the allowed visibility used for a group type.
+ *
+ * @param array $visibilities
+ *   The visibilities list.
+ * @param string $group_type_id
+ *   The Group type we alter the visibility setting for.
+ *
+ * @see social_group_get_allowed_visibility_options_per_group_type()
+ *
+ * @ingroup social_group_api
+ */
+function hook_social_group_allowed_visibilities_alter(array &$visibilities, $group_type_id) {
+  if ($group_type_id === 'custom_public_group') {
+    $visibilities['community'] = TRUE;
+  }
+}
+
 /**
  * Provide a method to alter default group overview route.
  *
  * @param array $route
  *   An array with route name and parameters.
- * @param GroupInterface $group
+ * @param \Drupal\group\Entity\GroupInterface $group
  *   Current group entity.
  *
  * @ingroup social_group_api
  */
-function hook_social_group_overview_route_alter(array &$route, GroupInterface $group) {
-  if ($group->bundle() == 'challenge') {
+function hook_social_group_overview_route_alter(array &$route, \Drupal\group\Entity\GroupInterface $group) {
```

Comment:
```suggestion
function hook_social_group_overview_route_alter(array &$route, GroupInterface $group) {
```

Add a use statement at the top of the file instead :)